In [20]:
pip install requests

Note: you may need to restart the kernel to use updated packages.


In [21]:
import requests
from bs4 import BeautifulSoup
from openpyxl import Workbook


In [22]:
thesis_dataset=Workbook()
thesis_s = thesis_dataset.active
thesis_s.append(['index', 'sub_seq', 'conference', 'year', 'title', 'author', 'abs'])


In [61]:
def add_aut(url):
    response = requests.get(url)
    soup = BeautifulSoup(response.content, "html.parser")

    aut = []
    temp = []

    div_1 = soup.select('#main > div')
    p = ''
    leng = len(div_1)
    leng2 = 0

    for i in range(leng):
        if (i == 0):
            continue
        p = div_1[i].select(".d-block")
        leng2 = len(p)
        for j in range(leng2):
            if (j == 0 or j == 1 or j == 2):
                continue
            for k in p[j].select(".d-block > a"):
                tmp = k.get_text()
                if (tmp == 'pdf\n' or tmp == 'bib\n' or tmp == 'abs' or tmp == '' or tmp == 'bib'):
                    continue
                temp.append(tmp)
            if (len(temp) != 0):
                aut.append(temp)
            temp = []

    #     print(len(aut))

    inputString = ''

    for i in range(len(aut)):
        inputString = ','.join(aut[i])
        thesis_s.cell(i + 7320, 6, inputString)
        inputString = ''
    thesis_dataset.save('C:/Users/a0102/Desktop/thesis_dataset.xlsx')

    print("author length : ", len(aut))

In [62]:
def crawler(url):
    response = requests.get(url)
    soup = BeautifulSoup(response.content, "html.parser")

    my_titles = soup.select(
        'p>span:nth-child(2)>strong>a'
    )

    tit = []
    for title in my_titles:
        tit.append(title.text)
    # print('tit',tit)

    abs = []
    abstracts = soup.findAll("div", {'class': 'card-body p-3 small'})
    for abstract in abstracts:
        abs.append(abstract.text)
    # print('abs',abs)


    print("title length : ", len(tit))
    # print("author length : ", len(aut))
    print("abstraction length : ", len(abs))

    s = []
    num = len(tit) - len(abs)
    ii = 0

    while True:
        try:
            s.append(soup.select('.card-body .list-pl-responsive li .align-middle')[ii].get_text())
            ii += 1
        except:
            break

    num = len(s)

    new_s = []
    for i in range(num):
        if i % 2 == 0:
            new_s.append(s[i])

    # for i in range(num):
    #    s.append(soup.select('.card-body .align-middle')[2*i].get_text())

    for t in tit:
        if t == 'Proceedings of the 18th BioNLP Workshop and Shared Task':
            tit.remove(t)
    # acl-2019가 위 항목을 아래의 for문 속 if t==ss에서 못 걸러내므로 따로 remove 해줌

    for t in tit:
        for ss in new_s:
            if t == ss:
                tit.remove(t)

    for i in range(len(abs)):
        thesis_s.cell(i + 7320, 3, 'eacl')
        # 7319부터 시작
    thesis_dataset.save('C:/Users/a0102/Desktop/thesis_dataset.xlsx')

    for i in range(len(abs)):
        thesis_s.cell(i + 7320, 4, 2017)
    thesis_dataset.save('C:/Users/a0102/Desktop/thesis_dataset.xlsx')

    for i in range(len(abs)):
        thesis_s.cell(i + 7320, 5, tit[i])
    thesis_dataset.save('C:/Users/a0102/Desktop/thesis_dataset.xlsx')

    for i in range(len(abs)):
        thesis_s.cell(i + 7320, 7, abs[i])
    thesis_dataset.save('C:/Users/a0102/Desktop/thesis_dataset.xlsx')

    add_aut(url)

    print("title length : ", len(tit))
    # print("author length : ", len(aut))
    print("abstraction length : ", len(abs))

acl 2020,2018,2019

In [27]:
crawler("https://www.aclweb.org/anthology/events/acl-2020/")

title length :  1235
abstraction length :  1213
author length :  1213
title length :  1213
abstraction length :  1213


In [30]:
crawler("https://www.aclweb.org/anthology/events/acl-2019/")

title length :  1449
abstraction length :  1422
author length :  1422
title length :  1422
abstraction length :  1422


In [33]:
crawler("https://www.aclweb.org/anthology/events/acl-2018/")

title length :  674
abstraction length :  654
author length :  654
title length :  654
abstraction length :  654


emnlp 19,18,17

In [36]:
crawler("https://www.aclweb.org/anthology/events/emnlp-2019/")

title length :  1122
abstraction length :  1103
author length :  1102
title length :  1103
abstraction length :  1103


In [39]:
crawler("https://www.aclweb.org/anthology/events/emnlp-2018/")

title length :  1024
abstraction length :  1008
author length :  1007
title length :  1008
abstraction length :  1008


In [42]:
crawler("https://www.aclweb.org/anthology/events/emnlp-2017/")

title length :  353
abstraction length :  351
author length :  350
title length :  351
abstraction length :  351


conll 19,18,17

In [45]:
crawler("https://www.aclweb.org/anthology/events/conll-2019/")

title length :  115
abstraction length :  113
author length :  113
title length :  113
abstraction length :  113


In [51]:
crawler("https://www.aclweb.org/anthology/events/conll-2018/") #마지막 덩어리 abs없음

title length :  101
abstraction length :  82
author length :  98
title length :  98
abstraction length :  82


In [54]:
crawler("https://www.aclweb.org/anthology/events/conll-2017/") #중간 덩어리 abs 없음

title length :  89
abstraction length :  74
author length :  86
title length :  86
abstraction length :  74


naacl 19,18,16

In [57]:
crawler("https://www.aclweb.org/anthology/events/naacl-2019/")

title length :  752
abstraction length :  729
author length :  729
title length :  729
abstraction length :  729


In [60]:
crawler("https://www.aclweb.org/anthology/events/naacl-2018/")

title length :  588
abstraction length :  569
author length :  569
title length :  569
abstraction length :  569


In [ ]:
crawler("https://www.aclweb.org/anthology/events/naacl-2016/") #abs 없음

eacl 17,14,12

In [63]:
crawler("https://www.aclweb.org/anthology/events/eacl-2017/")

title length :  290
abstraction length :  286
author length :  285
title length :  286
abstraction length :  286


In [ ]:
crawler("https://www.aclweb.org/anthology/events/eacl-2014/") #abs 없음

In [ ]:
crawler("https://www.aclweb.org/anthology/events/eacl-2012/") #abs 없음

acl-2019-------------------
title length :  1449 -> 1423 (-26)
author length :  0
abstraction length :  1422 
-> 문자열 직접 적어서 지움

conll-2018/-------------------
title length :  98
author length :  0
abstraction length :  82

conll-2017/-------------------
title length :  86
author length :  0
abstraction length :  74

naacl-2016/-------------------
abstraction length :  0

eacl-2014/---------------------
abstraction length :  0

eacl-2012/-------------------
abstraction length :  0